### PR TITLE
Feature/11702 configuration is updated all the time

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
CCLService can detect now if data came from cache or is default and won't indicate and update
Added missing unittests for CCLService

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11702

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift [hunk 1]
```diff
@@ -222,7 +222,17 @@ extension Service {
 		// Check if we have default value. If so, return it independent wich error we had
 		if let defaultModel = resource.defaultModel {
 			Log.info("Found some default value", log: .client)
-			return .success(defaultModel)
+			guard var modelWithMetadata = defaultModel as? MetaDataProviding else {
+				return .success(defaultModel)
+			}
+			Log.info("Found a defaultModel wich conforms to MetaDataProviding. Adding metadata now.", log: .client)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift:228]:
```suggestion
			Log.info("Found a defaultModel which conforms to MetaDataProviding. Adding metadata now.", log: .client)
```

File: src/xcode/ENA/ENA/Source/Services/CCLService/__tests__/CCLConfiguationTests.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	func cclConfigurationData() throws -> Data {
+		return try XCTUnwrap(
+			Archive.createArchiveData(
+				accessMode: .create,
+				cborData: HealthCertificateToolkit.CCLConfigurationCBORDataFake()
+			)
+		)
+	}
+
+	// MARk: - Configuration logic on CCLService Layer
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/CCLService/__tests__/CCLConfiguationTests.swift:21]:
```suggestion
	// MARK: - Configuration logic on CCLService Layer
```

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift [hunk 3]
```diff
@@ -69,34 +96,38 @@ class CCLService: CCLServable {
 		var boosterRulesDidUpdate: Bool = false
```
[the remark this answers is not in the document]
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift:96]: I think there is no tread-safety issue here, as these are local variables inside a function and not shared with anyone else, right? I think there is currently no scenario in which they could be read/written in parallel

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift:96]: Good point we might add a lock to make sure it becomes thread save. The dispatch group used is just handy to collect the requests made.

**Pull request merged**